pkg/operator/resource/resourcemerge/admissionregistration.go

```diff
@@ -0,0 +1,51 @@
+package resourcemerge
+
+import (
+	operatorsv1 "github.com/openshift/api/operator/v1"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// ExpectedMutatingWebhooksConfiguration returns last applied generation for MutatingWebhookConfiguration resource registered in operator
+func ExpectedMutatingWebhooksConfiguration(name string, previousGenerations []operatorsv1.GenerationStatus) int64 {
+	generation := GenerationFor(previousGenerations, schema.GroupResource{Group: admissionregistrationv1.SchemeGroupVersion.Group, Resource: "mutatingwebhookconfigurations"}, "", name)
+	if generation != nil {
+		return generation.LastGeneration
+	}
+	return -1
+}
+
+// SetMutatingWebhooksConfigurationGeneration updates operator generation status list with last applied generation for provided MutatingWebhookConfiguration resource
+func SetMutatingWebhooksConfigurationGeneration(generations *[]operatorsv1.GenerationStatus, actual *admissionregistrationv1.MutatingWebhookConfiguration) {
+	if actual == nil {
+		return
+	}
+	SetGeneration(generations, operatorsv1.GenerationStatus{
+		Group:          admissionregistrationv1.SchemeGroupVersion.Group,
+		Resource:       "mutatingwebhookconfigurations",
+		Name:           actual.Name,
+		LastGeneration: actual.ObjectMeta.Generation,
+	})
+}
+
+// ExpectedValidatingWebhooksConfiguration returns last applied generation for ValidatingWebhookConfiguration resource registered in operator
+func ExpectedValidatingWebhooksConfiguration(name string, previousGenerations []operatorsv1.GenerationStatus) int64 {
+	generation := GenerationFor(previousGenerations, schema.GroupResource{Group: admissionregistrationv1.SchemeGroupVersion.Group, Resource: "validatingwebhookconfigurations"}, "", name)
+	if generation != nil {
+		return generation.LastGeneration
+	}
+	return -1
+}
+
+// SetValidatingWebhooksConfigurationGeneration updates operator generation status list with last applied generation for provided ValidatingWebhookConfiguration resource
+func SetValidatingWebhooksConfigurationGeneration(generations *[]operatorsv1.GenerationStatus, actual *admissionregistrationv1.ValidatingWebhookConfiguration) {
+	if actual == nil {
+		return
+	}
+	SetGeneration(generations, operatorsv1.GenerationStatus{
+		Group:          admissionregistrationv1.SchemeGroupVersion.Group,
+		Resource:       "validatingwebhookconfigurations",
+		Name:           actual.Name,
+		LastGeneration: actual.ObjectMeta.Generation,
+	})
+}
```

Comment:
these functions here irritate me. Where will they be used?

The ones for deployments come from the fact that deployments often are the primary resources deployed by an operator. But webhook probably are not.

Reply:
Just updated the PR description with links and a code sample showing how it is supposed to be used. If the user does not require generation tracking and could tolerate multiple update calls to the server, just set the `expectedGeneration` to 0 like here: https://github.com/openshift/library-go/pull/836/files#diff-895b3186a55d12e7a602b6c33d4881042ac6f3aad924a271d3fabb3693236b0aR173

or use the generic apply method.